jaeger-core/src/test/java/io/jaegertracing/internal/JaegerSubclassTest.java

```diff
@@ -0,0 +1,254 @@
+          tags,
+          zipkinSharedRpcSpan,
+          scopeManager,
+          baggageRestrictionManager,
+          expandExceptionLogs,
+          objectFactory);
+    }
+  }
+
+  private static class CustomSpan extends JaegerSpan {
+    private CustomSpan(
+        CustomTracer tracer,
+        String operationName,
+        CustomSpanContext context,
+        long startTimeMicroseconds,
+        long startTimeNanoTicks,
+        boolean computeDurationViaNanoTicks,
+        Map<String, Object> tags,
+        List<Reference> references) {
+      super(
+          tracer,
+          operationName,
+          context,
+          startTimeMicroseconds,
+          startTimeNanoTicks,
+          computeDurationViaNanoTicks,
+          tags,
+          references);
+    }
+
+    @Override
+    public CustomSpanContext context() {
+      return (CustomSpanContext) super.context();
+    }
+  }
+
+  private static class CustomSpanContext extends JaegerSpanContext {
+    private CustomSpanContext(
+        long traceId,
+        long spanId,
+        long parentId,
+        byte flags,
+        Map<String, String> baggage,
+        String debugId,
+        CustomObjectFactory objectFactory) {
+      super(traceId, spanId, parentId, flags, baggage, debugId, objectFactory);
+    }
+
+    @Override
+    public CustomSpanContext withFlags(byte flags) {
+      return (CustomSpanContext) super.withFlags(flags);
+    }
+
+    @Override
+    public CustomSpanContext withBaggage(Map<String, String> baggage) {
+      return (CustomSpanContext) super.withBaggage(baggage);
+    }
+
+    @Override
+    public CustomSpanContext withBaggageItem(String key, String value) {
+      return (CustomSpanContext) super.withBaggageItem(key, value);
+    }
+  }
+
+  private static class CustomObjectFactory extends JaegerObjectFactory {
+    @Override
+    public CustomSpan createSpan(
+        JaegerTracer tracer,
+        String operationName,
+        JaegerSpanContext context,
+        long startTimeMicroseconds,
+        long startTimeNanoTicks,
+        boolean computeDurationViaNanoTicks,
+        Map<String, Object> tags,
+        List<Reference> references) {
+      return new CustomSpan(
+          (CustomTracer) tracer,
+          operationName,
+          (CustomSpanContext) context,
+          startTimeMicroseconds,
+          startTimeNanoTicks,
+          computeDurationViaNanoTicks,
+          tags,
+          references);
+    }
+
+    @Override
+    public CustomSpanContext createSpanContext(
+        long traceId,
+        long spanId,
+        long parentId,
+        byte flags,
+        Map<String, String> baggage,
+        String debugId) {
+      return new CustomSpanContext(traceId, spanId, parentId, flags, baggage, debugId, this);
+    }
+
+    @Override
+    public CustomTracer.CustomSpanBuilder createSpanBuilder(
+        JaegerTracer tracer, String operationName) {
+      return ((CustomTracer) tracer).new CustomSpanBuilder(operationName);
```

Comment:
`(CustomTracer) tracer)` - would `CustomTracer.this` work?

Reply:
In this context, `this` refers to the `CustomObjectFactory` instance, so I doubt it.